src/sentry/new_migrations/monkey/__init__.py

```diff
@@ -32,27 +32,26 @@
 # sure we're not missing any important changes.
 SENTRY_MIGRATION_TEMPLATE = """\
 %(imports)s
+from sentry.new_migrations.migrations import CheckedMigration
 
-class Migration(migrations.Migration):
-    # This flag is used to mark that a migration shouldn't be automatically run in
-    # production. We set this to True for operations that we think are risky and want
-    # someone from ops to run manually and monitor.
-    # General advice is that if in doubt, mark your migration as `is_dangerous`.
-    # Some things you should always mark as dangerous:
-    # - Large data migrations. Typically we want these to be run manually by ops so that
-    #   they can be monitored. Since data migrations will now hold a transaction open
-    #   this is even more important.
-    # - Adding columns to highly active tables, even ones that are NULL.
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
     is_dangerous = False
```

Comment:
Longer term I think we should work towards not needing to manually running dangerous migrations. We'll likely need a way for developers to self-serve run backfills/data migrations first though.

These changes are a great step towards that as we can use tooling to guide developers towards safe operations, and help them find ways to combine safe operations to do the unsafe thing they may need to accomplish.